### PR TITLE
Feature/163 form validation

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements: #163

### CODE REVIEW

**Comment by kevinwallimann** [comment]
image
Two problems here:
1. Wrong error message
2. Maxlength too small

Same for main class and value za.co.absa.hyperdrive.driver.drivers.CommandLineIngestionDriver

File: ui/src/app/components/workflows/workflow/jobs/job/job.component.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export class JobComponent implements OnInit, OnDestroy {
       this.selectedJob = !!selected ? selected.value : undefined;
     });
 
-    this.jobChangesSubscription = this.jobChanges.subscribe((jobChange) => {
+    this.jobChangesSubscription = this.jobChanges.pipe(delay(0)).subscribe((jobChange) => {
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow/jobs/job/job.component.ts:62]: Why is that needed? delay of zero seems redundant..

File: ui/src/app/stores/workflows/workflows.effects.ts [hunk 2]
```diff
@@ -298,13 +314,28 @@ export class WorkflowsEffects {
...
-        catchError(() => {
-          this.toastrService.error(texts.UPDATE_WORKFLOW_FAILURE_NOTIFICATION);
-          return [
-            {
-              type: WorkflowActions.UPDATE_WORKFLOW_FAILURE,
-            },
-          ];
+        catchError((errorResponse) => {
+          if (
+            errorResponse instanceof Array &&
```
Comment (kevinwallimann) [inline ui/src/app/stores/workflows/workflows.effects.ts]: You may want to extract this if-condition

File: ui/src/app/components/workflows/workflow/workflow.component.ts [hunk 3]
```diff
@@ -123,19 +139,32 @@ export class WorkflowComponent implements OnInit, OnDestroy {
   }
 
   createWorkflow() {
-    this.confirmationDialogServiceSubscription = this.confirmationDialogService
-      .confirm(ConfirmationDialogTypes.YesOrNo, texts.CREATE_WORKFLOW_CONFIRMATION_TITLE, texts.CREATE_WORKFLOW_CONFIRMATION_CONTENT)
-      .subscribe((confirmed) => {
-        if (confirmed) this.store.dispatch(new CreateWorkflow());
-      });
+    this.showHiddenParts();
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow/workflow.component.ts]: Maybe you could show hidden parts only if form was not valid? If the form is valid and the user presses cancel, the appearance will have changed for no apparent reason
Of course, you may also remove the confirmation box altogether here ;)

Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow/workflow.component.ts]: Fixed first part. I see you dont like confirmation box, but I will keep it there :D

File: ui/src/app/components/workflows/workflow/jobs/job/job.component.ts (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow/jobs/job/job.component.ts:62]: No otherwise we will get ExpressionChangedAfterItHasBeenCheckedError

File: ui/src/app/stores/workflows/workflows.effects.ts (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/stores/workflows/workflows.effects.ts]: Fixed

File: ui/src/app/components/workflows/workflow/jobs/job/job.component.ts (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow/jobs/job/job.component.ts:62]: > image
> Two problems here:
> 
> 1. Wrong error message
> 2. Maxlength too small
> 
> Same for main class and value za.co.absa.hyperdrive.driver.drivers.CommandLineIngestionDriver

Fixed

**Comment by jozefbakus** [comment]
> image
> Two problems here:
> 
> 1. Wrong error message
> 2. Maxlength too small
> 
> Same for main class and value za.co.absa.hyperdrive.driver.drivers.CommandLineIngestionDriver

Fixed

File: ui/src/app/components/workflows/workflow/dynamic-parts/guid-part/guid-part.component.html [hunk 4]
```diff
@@ -22,7 +22,7 @@
          [readOnly]="true"
          [ngModel]="value"
          (ngModelChange)="modelChanged($event)"
-         [size]="partValidationSafe.maxLength"
+         [size]="partValidationSafe.maxLength < 100 ? partValidationSafe.maxLength : 100"
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow/dynamic-parts/guid-part/guid-part.component.html:25]: Please extract the value 100 to a variable
Same in `key-string-value-part`, `string-part` and `string-sequence-part`

**Review by kevinwallimann — CHANGES_REQUESTED** [review]

**Comment by kevinwallimann** [comment]
Also I noticed that the symbol is gone for "Is active" in the show mode.
image

Is that intentional?

**Comment by jozefbakus** [comment]
> Also I noticed that the symbol is gone for "Is active" in the show mode.
> image
> 
> Is that intentional?

Yes, because it was easiest way how to do it without doing something with css.

File: ui/src/app/components/workflows/workflow/dynamic-parts/guid-part/guid-part.component.html (hunk 4 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow/dynamic-parts/guid-part/guid-part.component.html:25]: Fixed

**Review by kevinwallimann — APPROVED** [review]
